packages/core/src/barrel.ts

```diff
@@ -9,3 +9,4 @@ export * from "./keystore/index.js";
 export * from "./num/index.js";
 export * from "./signer/index.js";
 export * from "./utils/index.js";
+export * as codec from "./codec/index.js";
```

Comment:
Personally I think `pack` `unpack` `Packed` `Unpacked` `Packable` `Unpackable` are confusing. It looks like `Packed` is always `Bytes` here. Maybe we don't need this much flexibility.

I suggest to use
`encode` `decode` `DataType` instead, and encoder result should always be `Bytes`.
See https://en.wikipedia.org/wiki/Codec#:~:text=A%20codec%20is%20a%20device,a%20portmanteau%20of%20coder%2Fdecoder.

Reply:
confusing indeed there, because of it, I'm concerned to alter it, afraid of causing unexpected problems, however, speaking of which, I'd like to have a try fixing it.

Reply:
How about just changing them to:
Packed => Encoded
Packable => Encodable
Unpacked => Decoded
Unpackable => Decodable